pkg/controller/cassandradatacenter/controller/statefulset.go

```diff
@@ -0,0 +1,185 @@
+	}
+
+	// this is a deep copy
+	podSpec.Containers = containers
+
+	return podSpec
+}
+
+func newCassandraContainer(cdc *cassandraoperatorv1alpha1.CassandraDataCenter, dataVolumeClaim *corev1.PersistentVolumeClaim) *corev1.Container {
+	container := &corev1.Container{
+		Name:            "cassandra",
+		Image:           cdc.Spec.CassandraImage,
+		ImagePullPolicy: cdc.Spec.ImagePullPolicy,
+		Args:            []string{},
+		Ports: []corev1.ContainerPort{
+			{Name: "internode", ContainerPort: 7000},
+			{Name: "cql", ContainerPort: 9042},
+			{Name: "jmx", ContainerPort: 7199},
+		},
+		Resources: cdc.Spec.Resources,
+		SecurityContext: &corev1.SecurityContext{
+			Capabilities: &corev1.Capabilities{
+				Add: []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE"},
+			},
+		},
+		ReadinessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				Exec: &corev1.ExecAction{
+					Command: []string{"/usr/bin/cql-readiness-probe"},
+				},
+			},
+			InitialDelaySeconds: 60,
+			TimeoutSeconds:      5,
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: dataVolumeClaim.Name, MountPath: "/var/lib/cassandra"},
+		},
+	}
+
+	if cdc.Spec.PrometheusSupport == true {
+		container.Ports = append(container.Ports, corev1.ContainerPort{Name: "promql", ContainerPort: 9500})
+	}
+
+	return container
+}
+
+func addMountsToCassandra(cassandraContainer *corev1.Container, volumeMounts VolumeMounts) {
+	for _, vm := range volumeMounts {
+		cassandraContainer.VolumeMounts = append(cassandraContainer.VolumeMounts, corev1.VolumeMount{
+			Name:      vm.Volume.Name,
+			MountPath: vm.MountPath,
+		})
+
+		// entrypoint takes mount paths as arguments
+		cassandraContainer.Args = append(cassandraContainer.Args, vm.MountPath)
+	}
+}
+
+func addMountsToSidecar(sidecarContainer *corev1.Container, volumeMounts VolumeMounts) {
+	for _, vm := range volumeMounts {
+		// provide access to config map volumes in the sidecar, these reside in /tmp though and are not overlayed into /etc/cassandra
+		// TODO: rework this
+		sidecarContainer.VolumeMounts = append(sidecarContainer.VolumeMounts, corev1.VolumeMount{
+			Name:      vm.Volume.Name,
+			MountPath: vm.MountPath,
+		})
+	}
+}
+
+func addMountsToPodSpec(podSpec *corev1.PodSpec, volumeMounts VolumeMounts) {
+	for _, vm := range volumeMounts {
+		podSpec.Volumes = append(podSpec.Volumes, vm.Volume)
+	}
+}
+
+func newSidecarContainer(cdc *cassandraoperatorv1alpha1.CassandraDataCenter, dataVolumeClaim *corev1.PersistentVolumeClaim, podInfoVolume *corev1.Volume) *corev1.Container {
+
+	return &corev1.Container{
+		Name:            "sidecar",
+		Image:           cdc.Spec.SidecarImage,
+		ImagePullPolicy: cdc.Spec.ImagePullPolicy,
+		Ports: []corev1.ContainerPort{
+			{Name: "http", ContainerPort: 4567},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: dataVolumeClaim.Name, MountPath: "/var/lib/cassandra"},
+			{Name: podInfoVolume.Name, MountPath: "/etc/pod-info"},
+		},
+	}
+}
+
+func getPodInfoVolume() *corev1.Volume {
```

Comment:
`new`

Reply:
Since this function just returns a struct that is in no-way parameterised, perhaps its better to declare it as a package-local `var`? (Can those be `const`/`static`?)

Reply:
are you sure it wont be parametrised? I would leave it as it is, plus we have other "new" methods so I dont see the point to having it like var just because it does not accept any arguments.